### PR TITLE
iserver-test: Add 9.2 test and some improvements

### PR DESCRIPTION
**Pull request by ericywl**
## Motivation/summary

- Add 9.2 upgrade paths manually, since we don't have https://github.com/elastic/apm-server/issues/17602 yet
- Add input option in workflow dispatch to control which tests to run
- Add retry for `telemetrygen` in case of connection issues during ingestion
- Remove flush trigger since it seems to interfere with metrics aggregation sometimes

## How to test these changes

Run workflow: https://github.com/elastic/apm-server/actions/runs/16346566007

### CODE REVIEW

File: .github/workflows/integration-server-test.yml [hunk 1]
```diff
@@ -31,18 +47,17 @@ jobs:
         uses: ./.github/workflows/generate-bc-upgrade-paths
 
   run-upgrade:
-    name: Upgrade tests (Snapshot)
+    if: ${{ inputs.run-upgrade-tests == true }}
```
Comment (v1v) [inline .github/workflows/integration-server-test.yml]: I recall some edge cases when using `workflow_dispatch` default inputs and the `schedule`. Can you double-check if the default values are honoured when the schedule event runs?

I did run a quick search and found this [StackOverflow thread](https://stackoverflow.com/a/73495922) or [this](https://github.com/orgs/community/discussions/137750)

Comment (ericywl) [reply .github/workflows/integration-server-test.yml]: I see, will try another way to do it instead. Thanks for the heads up!

**Review by simitt — APPROVED** [review]
